app/src/main/java/org/piwigo/helper/URLHelper.java

```diff
@@ -0,0 +1,79 @@
+package org.piwigo.helper;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.os.AsyncTask;
+
+import org.piwigo.ui.login.LoginActivity;
+import org.piwigo.ui.main.MainActivity;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLPeerUnverifiedException;
+
+public class URLHelper extends AsyncTask<String, Void, String> {
+
+    @Override
+    protected String doInBackground(String... url) {
+        url[0] = url[0].replaceAll("https://", "").replaceAll("http://", "");
+        try {
+            if (isHttpsWebsite(url[0]))
+                return ("https://" + url[0]);
+            else if (isHttpWebsite(url[0]))
+                return ("http://" + url[0]);
+            else
+                return (null);
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+            return (null);
+        }
+    }
+
+    private boolean isHttpsWebsite(String siteUrl) throws IOException
```

Comment:
really throw an IOException here? - You catch SSLException, so why not IOException?